build_resolvers/lib/src/resolver.dart

```diff
@@ -152,9 +151,9 @@ class AnalyzerResolver implements ReleasableResolver {
   @override
   Future<bool> isLibrary(AssetId assetId) async {
     var source = _driver.sourceFactory.forUri2(assetId.uri);
-    return source != null &&
-        source.exists() &&
-        (await _driver.getSourceKind(assetPath(assetId))) == SourceKind.LIBRARY;
+    if (source == null || !source.exists()) return false;
+    var result = _driver.getFileSync2(assetPath(assetId));
+    return result is FileResult && !result.isPart;
```

Comment:
one thing I was a little worried about is changing behavior subtly in cases where we get an invalid result.

If we introduce a bug down the line, will we start treating some libraries and non-libraries when the real situation is using the analyzer incorrectly?

Reply:
Do we already know that non-Dart files won't show up here? Could a non-part, non-library file show up as if it was a library?

cc @scheglov - this is similar to the internal code so the same concerns likely apply.